### PR TITLE
Fix list in progress backup panic

### PR DESCRIPTION
**Pull request by jenting**
#### Proposal Change
Returns error at in-progress backup, so we could check do ignore it _or_ include it when all the jobs finished.
Add unit test to cover the in-progress backup and faulty backup cases.

#### Linked Issue
https://github.com/longhorn/longhorn/issues/2683

Signed-off-by: JenTing Hsiao <jenting.hsiao@suse.com>

### CODE REVIEW

File: list_test.go [hunk 1]
```diff
@@ -228,7 +243,7 @@ func TestListSingleVolumeBackups(t *testing.T) {
 	for volumeName := range m.volumeInfos {
 		volumeInfo, err := List(volumeName, mockDriverURL, false)
 		assert.NoError(err)
-		assert.Equal(m.numOfBackups, len(volumeInfo[volumeName].Backups))
+		assert.Equal(m.numOfBackups-m.numOfInProgressBackups, len(volumeInfo[volumeName].Backups))
```
Comment (shuo-wu) [inline list_test.go:246]: Should we check faulty backups as well?

File: list.go [hunk 2]
```diff
@@ -108,6 +108,7 @@ func addListVolume(volumeName string, driver BackupStoreDriver, volumeOnly bool)
 				info = failedBackupInfo(backupName, volumeName, driver.GetURL(), err)
 			} else if isBackupInProgress(backup) {
 				// for now we don't return in progress backups to the ui
+				return nil, fmt.Errorf("Backup in progress")
```
[the remark this answers is not in the document]
Comment (jenting) [reply list.go:111]: I'd want to make it the same as https://github.com/longhorn/backupstore/blob/89bd8d2/list.go#L255

File: list_test.go (hunk 1 above)
Comment (jenting) [reply list_test.go:246]: Good suggestion!

File: list.go [hunk 3]
```diff
@@ -118,7 +119,10 @@ func addListVolume(volumeName string, driver BackupStoreDriver, volumeOnly bool)
 	}
 
 	for _, tracker := range trackers {
-		payload, _ := tracker.Result()
+		payload, err := tracker.Result()
+		if err != nil {
+			continue
```
Comment (innobead) [inline list.go:124]: Can we at least dump the error log here? debug or trace.

**Review by innobead — APPROVED** [review]
LGTM

**Pull request merged**